tests/asset-store/pkg/waiter/waiter.go

```diff
@@ -0,0 +1,26 @@
+package waiter
+
+import (
+	"fmt"
+	"log"
+	"time"
+)
+
+func WaitAtMost(fn func() (bool, error), duration time.Duration) error {
```

Comment:
I know that we have such function also in other projects. It should be moved to common

Reply:
I absolutely agree. However, I would do it as a separate PR.